virtcontainers/agent.go

```diff
@@ -221,4 +221,7 @@ type agent interface {
 
 	// reseedRNG will reseed the guest random number generator
 	reseedRNG(data []byte) error
+
+	// hotplugCommunicationChannel hot plugs the communication channel in the hypervisor
+	hotplugCommunicationChannel(h hypervisor) error
```

Comment:
I was thinking we could have a more generic function here, something like `waitForAgentReady()` or `completeAgentInitialization()`.
This would allow several different actions to be gathered and handled by this same function, and from a higher perspective (`sandbox.go`), I think the only thing we want to make sure is that both VM and agent are ready before we can proceed further.
```go
waitForAgentReady(s *Sandbox) error
```

Reply:
Agree, lets make this api more generic. `waitForAgentReady()` sounds good to me. 
Agents that dont need to do additional steps can simply return true.